lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/serde/GeometrySerde.java

```diff
@@ -286,21 +265,12 @@ public static Envelope deserializeEnvelope(Slice shape)
 
     private static Envelope getEnvelope(BasicSliceInput input, GeometrySerializationType type, int length)
     {
-        switch (type) {
-            case POINT:
-                return getPointEnvelope(input);
-            case MULTI_POINT:
-            case LINE_STRING:
-            case MULTI_LINE_STRING:
-            case POLYGON:
-            case MULTI_POLYGON:
-                return getSimpleGeometryEnvelope(input, length);
-            case GEOMETRY_COLLECTION:
-                return getGeometryCollectionOverallEnvelope(input);
-            case ENVELOPE:
-                return readEnvelope(input);
-        }
-        throw new IllegalArgumentException("Unexpected type: " + type);
+        return switch (requireNonNull(type, "geometrySerializationType is null")) {
```

Comment:
why `requireNonNull` ? it wasn't there before your change.

btw

```
requireNonNull(type, "geometrySerializationType is null")
```

should be 

```
requireNonNull(type, "type is null")
```